kernel/src/hil/hmac.rs

```diff
@@ -0,0 +1,39 @@
+//! Interface for HMAC
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    // On success the `result` will contain the digest. On error it will contain
+    // the HMAC error value.
+    fn hash_done(&'a self, result: Result<&'a mut [u8], u32>);
```

Comment:
Probably worth making the error types an enum, even if the errors are specific to this hardware implementation.

Reply:
The problem is that I can't find a list of the possible errors.